File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/item/ItemDao.java
package com.legue.axel.lolsummonertool.database.dao.item;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.legue.axel.lolsummonertool.database.model.item.Item;

import java.util.List;

@Dao
public interface ItemDao {
    @Query("SELECT * FROM items ORDER BY name")
    LiveData<List<Item>> getItems();

    @Query("SELECT * FROM items WHERE id = :itemId")
    LiveData<Item> getItemById(int itemId);

    @Insert
    void insertItem(Item item);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllItem(List<Item> items);

    @Delete
    void deleteItem(Item champion);

    @Query("DELETE FROM items")
    void deleteAll();

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateItem(Item item);
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/item/ItemDao.kt
package com.legue.axel.lolsummonertool.database.dao.item

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update

import com.legue.axel.lolsummonertool.database.model.item.Item

@Dao
interface ItemDao {
    @Query("SELECT * FROM items ORDER BY name")
    fun getItems(): LiveData<List<Item>>

    @Query("SELECT * FROM items WHERE id = :itemId")
    fun getItemById(itemId: Int): LiveData<Item>

    @Insert
    fun insertItem(item: Item)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllItem(items: List<Item>)

    @Delete
    fun deleteItem(champion: Item)

    @Query("DELETE FROM items")
    fun deleteAll()

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateItem(item: Item)
}
